src/components/CartToggleButton/CartToggleButton.tsx

```diff
@@ -96,34 +96,36 @@ const CartToggleButton = ({
   basketId,
   isNotBasketCountMAX,
   timeoutRef,
+  isSoldOut,
 }: CartToggleButtonProps) => {
   const imageSrc = isInBascket
     ? IMAGE_PATH.SHOPPIN_CART_REMOVE
     : IMAGE_PATH.SHOPPIN_CART_ADD;
-  const { fetchCartItems } = useCartContext();
-  const { setError, setErrorMessage } = useUIContext();
+  const { fetchCartItems, setError, setErrorMessage } = useDataContext();
 
   return (
-    <CartToggleButtonWrapper
-      isInBascket={isInBascket}
-      onClick={() =>
-        handleCartToggleButton({
-          isInBascket,
-          productId: id,
-          basketId,
-          isNotBasketCountMAX,
-          timeoutRef,
-          fetchCartItems,
-          setError,
-          setErrorMessage,
-        })
-      }
-    >
-      <img src={imageSrc} alt="shopping_cart" />
-      <CartToggleButtonText isInBascket={isInBascket}>
-        {isInBascket ? "빼기" : "담기"}
-      </CartToggleButtonText>
-    </CartToggleButtonWrapper>
+    !isSoldOut && (
```

Comment:
취향이지만 이런 식으로 작성도 가능하겠네요 ㅎㅎ

```suggestion
    if (!isSoldOut) return null;
    
    return (
       <CartToggleButtonWrapper />
    )
```